### PR TITLE
feat(ContentBlockHeadlines): add ContentBlockHeadlines pattern

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#3007 

### Description

Adds new `ContentBlockHeadlines` pattern

### Changelog

**New**

- `ContentBlockHeadlines` pattern
- `ContentBlockHeadlines` styles, story, and documentation

### CODE REVIEW

File: packages/react/src/patterns/blocks/ContentBlockHeadlines/__stories__/ContentBlockHeadlines.stories.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Copyright IBM Corp. 2016, 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { object, text } from '@storybook/addon-knobs';
+import ContentBlockHeadlines from '../ContentBlockHeadlines';
+import { DDS_CONTENTBLOCK_HEADLINES } from '../../../../internal/FeatureFlags';
+import items from './data/items.json';
```
Comment (jeffchew) [inline packages/react/src/patterns/blocks/ContentBlockHeadlines/__stories__/ContentBlockHeadlines.stories.js:11]: @kennylam we should probably get out of the habit of json blocks for knobs, any way to break this down?

**Comment by ltitle** [comment]
@kennylam 3 small changes:

1. The end of section horizontal rule should span the full 12 columns not just the 8 columns like the in section rules above

2. Top of section padding on Medium breakpoint should be 32px. Is currently showing 64px:

image

3. On small breakpoint the spacing at the top of each item should remain 16px (same as all other breakpoints). It is currently increasing to 24px:

image

File: packages/react/src/patterns/blocks/ContentBlockHeadlines/__stories__/ContentBlockHeadlines.stories.js (hunk 1 above)
Comment (kennylam) [reply packages/react/src/patterns/blocks/ContentBlockHeadlines/__stories__/ContentBlockHeadlines.stories.js:11]: @jeffchew I've been of the opinion for a while that we shouldn't include those objects as knobs at all. Especially with our plans to create Codesandbox examples for everything, it's not needed. I'd say we just remove them entirely.

Comment (jeffchew) [reply packages/react/src/patterns/blocks/ContentBlockHeadlines/__stories__/ContentBlockHeadlines.stories.js:11]: As discussed, we should either have dynamic item knobs like http://ibmdotcom-react.mybluemix.net/?path=/story/components-buttongroup--default, or remove for now and we address later.

Comment (kennylam) [reply packages/react/src/patterns/blocks/ContentBlockHeadlines/__stories__/ContentBlockHeadlines.stories.js:10]: @jeffchew Removing for now. Will revisit the story later and look into adding dynamic items that can have randomized content to test `sameHeight` and CTA or not.

**Pull request merged**